lib/galaxy/security/validate_user_input.py

```diff
@@ -75,11 +84,23 @@ def validate_email(trans, email, user=None, check_dup=True, allow_empty=False):
         domain = extract_domain(email, base_only=True)
         if domain in trans.app.config.email_domain_blocklist_content:
             message = "Please enter your permanent email address."
+    if not message and validate_domain:
+        domain = email.split('@', 1)[1]
+        message = validate_domain(domain)
+    return message
+
+
+def validate_domain(domain):
+    message = None
+    try:
+        socket.gethostbyname(domain)
+    except socket.gaierror:
+        message = "The email domain cannot be resolved."
     return message
 
 
 def extract_domain(email, base_only=False):
-    domain = email.split('@')[1]
+    domain = email.split('@', 1)[1]
```

Comment:
Since this awesome PR contains such awesome tests, could we also have a test for this change? Something like this
`assert extract_domain('a@a.b@.com') == 'a.b@.com'`?

Reply:
I could, but `a@a.b@.com` wouldn't validate as an address since `@` is not in `\w`, so although it passes `extract_domain()`, it should never get there, and `a.b@.com` is not a valid domain.

Technically a quoted `@` can appear in the local part of an address, but anyone doing that is surely just trying to break applications and mail systems.

Reply:
I guess this should be:
```suggestion
    domain = email.rsplit('@', 1)[-1]
```
to be the Correctest™.

Reply:
But then what's the point of limiting the # of splits in the function? After all, it's a top-level public function, and the new code guards against precisely this type of input. (I'm sorry, this is total bikeshedding! I'm 100% +1 either way!)

Reply:
> But then what's the point of limiting the # of splits in the function?
RAAAAAAAAAAAAAAAAAAW SPEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEED

Reply:
```
import time
def test_rawwwwwww_speeeeeeeed_in_a_very_non_flaky_way():
    start = time.perf_counter_ns()
    'a@a@a'.split('@')
    end = time.perf_counter_ns()
    delta1 = end - start
    start = time.perf_counter_ns()
    'a@a@a'.split('@', 1)
    end = time.perf_counter_ns()
    delta2 = end - start
    assert delta2 < delta1
```